charmcraft/commands/store/__init__.py

```diff
@@ -583,3 +608,54 @@ def _get_lib_info(*, full_name=None, lib_path=None):
     return LibData(
         lib_id=libid, api=libapi, patch=libpatch, content_hash=content_hash, content=content,
         full_name=full_name, path=lib_path, lib_name=lib_name, charm_name=charm_name)
+
+
+class CreateLibCommand(BaseCommand):
+    """Create a charm library."""
+    name = 'create-lib'
+    help_msg = "Create a charm library."
+    overview = textwrap.dedent("""
+        Create a charm library.
+
+        It will request a unique ID from Charmhub and bootstrap a
+        template file in the proper local directory.
+
+        It will automatically take you through the login process if
+        your credentials are missing or too old.
+    """)
+
+    def fill_parser(self, parser):
+        """Add own parameters to the general parser."""
+        parser.add_argument(
+            'lib_name', metavar='lib-name',
+            help="The name of the library file (e.g. 'db').")
+
+    def run(self, parsed_args):
+        """Run the command."""
+        lib_name = parsed_args.lib_name
+        valid_chars = set(string.ascii_lowercase + string.digits + '_')
+        if set(lib_name) - valid_chars or lib_name[0] not in string.ascii_lowercase:
+            raise CommandError(
+                "Invalid library name (can be only lowercase alphanumeric "
+                "characters and underscore, starting with alpha).")
```

Comment:
I added a check for the `""` name, which is obviously incorrect, but not for a "too long" situation. However, I started handling (with its test!) a more generic problem of not being able to write the library (because of name too long, or permissions, disk full, or whatever may happen there).